src/utils/changeChain.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  internalAddressesPerNetwork,
+  ValidNetworkNameTypes,
+} from "../utils/contracts/contractAddresses/internalAddresses";
+import { injectedConnector } from "../hooks/injectedConnectors";
+
+function switchEthereumChain(connector: any, chainId: number) {
+  connector.request({
```

Comment:
This creates a `Promise`, which is then immediately forgotten. There's no way to actually monitor whether or not it succeeded, but the console will surely yell about it.

If you invoke an `async` function, you should `await` it from within an `async` function or return the `Promise` it produced.

Reply:
I'm going to correct the Promise semantics in my own commit to your PR branch. Marking as resolved.